src/explode.c

```diff
@@ -567,11 +568,16 @@ explode(
                 else if (str != g.killer.name && str != hallu_buf)
                     Strcpy(g.killer.name, str);
                 g.killer.format = KILLED_BY_AN;
+            } else if (olet == TRAPPED_DOOR || olet == BURNING_OIL) {
+                g.killer.format = KILLED_BY_AN;
+                Snprintf(g.killer.name, sizeof g.killer.name,
+                         "exploding %s",
+                         olet == BURNING_OIL ? "fire bomb" : "door");
```

Comment:
You can catch yourself in your own burning oil though, in which case the hero should be blamed in the death. "caught themself in their own fire bomb" or something similar. This could probably be differentiated with a g.mon_moving check here?